core/monitor/price/pricemonitoring.go

```diff
@@ -446,6 +446,8 @@ func (e *Engine) checkBounds(trades []*types.Trade) []*types.PriceMonitoringTrig
 				ret = append(ret, b.Trigger)
 				// deactivate the bound that just got violated so it doesn't prevent auction from terminating
 				b.Active = false
+				// only allow breaking one bound at a time
```

Comment:
This is fine IMO, but I believe back when this was first implemented, the auction duration was supposed to depend on by how much the price had moved out of the bounds, hence we checked all price bounds, and to me, it looks like the spec still reflects this: https://github.com/vegaprotocol/specs/blob/master/protocol/0032-PRIM-price_monitoring.md#view-from-the-price-monitoring-engine-side specifically the bit that reads: _"to give an example, with 3 triggers the price protection auction can be calculated as follows"_, where 3 examples are listed, with an auction of 5m, 15m, and 1h respectively depending on how much the price bound was exceeded